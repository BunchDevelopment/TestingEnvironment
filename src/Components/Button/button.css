/* default button style */
.btnStyle {
  margin: 5px;
}

/* Button Types */
.primary {
  color: #fff;
  background-color: #337ab7;
  border-color: #2e6da4;
  border: 1px solid transparent;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.3);
}
.primary:hover {
  background-color: #255c8b;
}
.primary:disabled {
  background-color: #7ba8d0;
  color: white;
}
.success {
  color: #fff;
  background-color: #5cb85c;
  border-color: #4cae4c;
  border: 1px solid transparent;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.3);
}
.success:hover {
  background-color: #387038;
}
.success:disabled {
  background-color: #7dcf7d;
  color: white;
}
.warning {
  color: #fff;
  background-color: #f0ad4e;
  border-color: #eea236;
  border: 1px solid transparent;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.3);
}
.warning:hover {
  background-color: #b9853c;
}
.warning:disabled {
  background-color: #f5ca8c;
  color: white;
}

.danger {
  color: #fff;
  background-color: #d9534f;
  border-color: #d43f3a;
  border: 1px solid transparent;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.3);
}
.danger:hover {
  background-color: #a33f3b;
}
.danger:disabled {
  background-color: #e48e8b;
  color: white;
}
/* button sizes */
.xsmall {
  padding: 1px 5px;
  font-size: 12px;
  line-height: 1.5;
  border-radius: 3px;
  min-width: 40px;
}
.small {
  min-width: 100px;
  height: 25px;
  padding: 2px 10px 4px 10px;
  font-size: 14px;
  border-radius: 6px;
}
.medium {
  min-width: 125px;
  height: 30px;
  padding: 2px 12px 6px 12px;
  font-size: 16px;
  border-radius: 6px;
}
.large {
  min-width: 150px;
  height: 35px;
  padding: 8px 14px 8px 14px;
  font-size: 18px;
  border-radius: 6px;
}
