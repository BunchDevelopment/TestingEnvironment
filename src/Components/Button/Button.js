import React from "react";
import "./button.css";

const TYPES = {
  PRIMARY: "primary",
  WARNING: "warning",
  DANGER: "danger",
  SUCCESS: "success"
};

export const SIZES = {
  XS: "xsmall",
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large"
};

const BaseButton = ({
  onClick,
  type,
  disabled,
  buttonType,
  buttonSize,
  title,
  styles,
  children,
  testid
}) => (
  <button
    type={type}
    disabled={disabled}
    onClick={onClick}
    title={title}
    style={styles}
    data-testid={testid}
    // quick ternary to remove style for default disabled style
    className={`${buttonType} ${buttonSize} btnStyle`}
  >
    {children}
  </button>
);

export const Primary = props => (
  <BaseButton {...props} buttonType={TYPES.PRIMARY} />
);
export const Warning = props => (
  <BaseButton {...props} buttonType={TYPES.WARNING} />
);
export const Danger = props => (
  <BaseButton {...props} buttonType={TYPES.DANGER} />
);
export const Success = props => (
  <BaseButton {...props} buttonType={TYPES.SUCCESS} />
);
