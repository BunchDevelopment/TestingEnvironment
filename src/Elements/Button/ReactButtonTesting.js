import React from "react";
import { Primary } from "./Button";

//Created to create a button to test state and incrementing state with  button
class TestingButton extends React.Component {
  state = {
    count: 0
  };

  count = () => {
    this.setState(prevState => ({
      count: prevState.count + 1
    }));
  };

  render() {
    const { count } = this.state;
    return (
      <>
        {/* <Primary testid="testOne" buttonSize="medium" onClick={this.count}>
          {count}
        </Primary>
        <Primary testid="test2" buttonSize="medium" onClick={this.count}>
          {count}
        </Primary> */}
      </>
    );
  }
}

export default TestingButton;
