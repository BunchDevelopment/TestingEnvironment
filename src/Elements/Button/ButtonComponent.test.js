import React from "react";
import { render, cleanup } from "@testing-library/react";
import * as Button from "./Button";

afterEach(cleanup);
//Checks snapshot of button components
test("Button Component Snapshot", () => {
  const { container, debug } = render(
    <>
      <Button.Primary />
      <Button.Warning />
      <Button.Danger />
      <Button.Success />
    </>
  );
  debug();
  expect(container.children).toMatchSnapshot();
});
