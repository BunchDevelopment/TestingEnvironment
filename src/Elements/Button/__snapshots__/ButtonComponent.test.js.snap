// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshot 1`] = `
HTMLCollection [
  <button
    class="primary undefined btnStyle"
  />,
  <button
    class="warning undefined btnStyle"
  />,
  <button
    class="danger undefined btnStyle"
  />,
  <button
    class="success undefined btnStyle"
  />,
]
`;
